tests/sympc/tensor/replicatedshare_tensor_test.py

```diff
@@ -371,3 +384,214 @@ def test_rshift() -> None:
+    y_share = ReplicatedSharedTensor(shares=[y], ring_size=ring_size)
+
+    expected_res = prime_op(x, y)
+    result = op(x_share, y_share)
+    result = result.shares[0]
+
+    assert (result == expected_res).all()
+
+
+@pytest.mark.parametrize("op_str", ["add", "sub"])
+def test_ops_prime_share_public(op_str) -> None:
+    op = getattr(operator, op_str)
+    ring_size = PRIME_NUMBER
+    prime_op = ReplicatedSharedTensor.get_op(ring_size, op_str)
+    x = torch.tensor([[24, 34], [66, 1]], dtype=torch.uint8)
+    y = torch.tensor([[34, 47], [45, 32]], dtype=torch.uint8)
+
+    x_share = ReplicatedSharedTensor(shares=[x], ring_size=ring_size)
+
+    expected_res = prime_op(x, y)
+    result = op(x_share, y)
+    result = result.shares[0]
+
+    assert (result == expected_res).all()
+
+
+@pytest.mark.parametrize("security", ["semi-honest", "malicious"])
+def test_ops_prime_public_mul(get_clients, security) -> None:
+    parties = get_clients(3)
+    protocol = Falcon(security)
+    session = Session(protocol=protocol, parties=parties)
+    SessionManager.setup_mpc(session)
+    ring_size = PRIME_NUMBER
+    bin_op = ReplicatedSharedTensor.get_op(ring_size, "mul")
+
+    sh = torch.tensor([[33, 45, 0], [52, 41, 22]], dtype=torch.uint8)
+    shares = [sh, sh, sh]
+    rst_list = ReplicatedSharedTensor.distribute_shares(
+        shares=shares, session=session, ring_size=ring_size
+    )
+    tensor = MPCTensor(shares=rst_list, session=session)
+    tensor.shape = sh.shape
+
+    secret = ReplicatedSharedTensor.shares_sum(shares, ring_size)
+
+    value = torch.tensor([66], dtype=torch.uint8)
+
+    result = operator.mul(tensor, value)
+    expected_res = bin_op(secret, value)
+
+    assert (result.reconstruct(decode=False) == expected_res).all()
+
+
+@pytest.mark.parametrize("op_str", ["add", "sub", "mul"])
+def test_ops_mod_prime(op_str) -> None:
+    op_prime = getattr(ReplicatedSharedTensor, op_str + "modprime")
+    op = getattr(operator, op_str)
+
+    input_lst = [val for val in range(PRIME_NUMBER)]
+
+    for val1, val2 in itertools.combinations(input_lst, 2):
+        tensor1 = torch.tensor([val1], dtype=torch.uint8)
+        tensor2 = torch.tensor([val2], dtype=torch.uint8)
+
+        assert op(val1, val2) % PRIME_NUMBER == op_prime(tensor1, tensor2)
+
+
+@pytest.mark.parametrize("op_str", ["add", "sub", "mul"])
+def test_exception_prime_dtype(op_str) -> None:
+    a = torch.tensor([1], dtype=torch.int8)
+    b = torch.tensor([1], dtype=torch.int16)
+
+    op = getattr(ReplicatedSharedTensor, op_str + "modprime")
+
+    with pytest.raises(ValueError):
+        op(a, b)
+
+
+def test_exception_get_op() -> None:
+
+    with pytest.raises(ValueError):
+        ReplicatedSharedTensor.get_op(28, "add")
+
+
+@pytest.mark.parametrize("security", ["semi-honest", "malicious"])
+@pytest.mark.parametrize("bit", [0, 1])
+def test_ops_bin_public_xor(get_clients, security, bit) -> None:
```

Comment:
MAYBE : could we also have tests where we do `xor` between two MPCTensors in `mpc_tensor_test.py` with ring_size 2,PRIME_NUMBER,session ring size.
Having the MPCTensors as shares of a bit in the corresponding `ring_size`.